src/components/Editor/Project/Project.js

```diff
@@ -108,49 +82,44 @@ const Project = (props) => {
         }
       }
     }, 2000);
-     
+
     return () => clearTimeout(debouncer)
-  }, [dispatch, forWebComponent, project, user])
+  }, [dispatch, project, user])
+
+  const openFileTab = (fileName) => {
+    if (openFiles.includes(fileName)) {
+      switchToFileTab(openFiles.indexOf(fileName), fileName)
+    } else {
+      dispatch(openFile(fileName))
+      switchToFileTab(openFiles.length)
+    }
+  }
+
+  const switchToFileTab = (index) => {
+    dispatch(setFocussedFileIndex(index))
+  }
 
   return (
-    <div className='proj'>
-      <div className={`proj-container${forWebComponent ? ' proj-container--wc': ''}`}>
-      {!forWebComponent ? <SideMenu openFileTab={openFileTab}/> : null}
-        <div className='proj-editor-container'>
-          <Tabs selectedIndex={focussedFileIndex} onSelect={index => switchToFileTab(index)}>
-            <div className='react-tabs__tab-container'>
-              <TabList>
-                {openFiles.map((fileName, i) => (
-                  <Tab key={i}>
-                    <span
-                      className={`react-tabs__tab-inner${fileName !== 'main.py'? ' react-tabs__tab-inner--split': ''}`}
-                      ref={tabRefs.current[project.components.findIndex(file => `${file.name}.${file.extension}`===fileName)]}>
-                        {fileName}
-                        {fileName !== 'main.py' ?
-                          <Button className='btn--tertiary react-tabs__tab-inner-close-btn' label='close' onClickHandler={(e) => closeFileTab(e, fileName)} ButtonIcon={() => <CloseIcon scaleFactor={0.85}/> }/>
-                        : null
-                        }
-                    </span>
-                  </Tab>
-                ))}
-              </TabList>
-            </div>
-            {openFiles.map((fileName, i) => (
-              <TabPanel key={i}>
-                <EditorPanel fileName={fileName.split('.')[0]} extension={fileName.split('.').slice(1).join('.')} />
-              </TabPanel>
-            ))}
-            <RunnerControls />
-          </Tabs>
+    <>
+      { !loading && data ?
+        <>
+          { isEmbedded ? null : <Header projectHeaderData={headerData} /> }
+        </>
+        : null
+      }
+      { loading ? <p>{t('project.loading')}</p> : null }
+      { error ? <p>{t('project.loadingFailed')}</p> : null }
```

Comment:
I'm not sure... Will this not still show the rest of the `Project` component in the loading and failed states? Or am I parsing it wrong? 🤔

Reply:
Ohhhh yes, I forgot about this, yes, that's fine for now then 👍